docs/development/services.md

```diff
@@ -0,0 +1,76 @@
+# Services
+## Network Diagram
+![Network Diagram](network_diagram.png)
+
+
+## Grapl Services - main pipeline
+Unless otherwise specified, the input to a service is the output of the one described above it.
+
+
+### Graph Generator
+**Input:** Event logs (i.g. sysmon logs, osquery logs, Cloudtrail logs) from the customer's S3 bucket.
+
+**Work:** Turns these events into a standalone subgraph, independent of existing Dgraph state.
+
+**Output:** Stores the subgraph to S3 for Node Identifier.
+
+
+### Node Identifier
+**Work:** Identifies nodes in the incoming subgraph against DynamoDB. 
+
+For instance, an incoming subgraph may refer to a process `{PID:1234, name: "coolthing.exe", started at 1:00AM}`; it's possible that Dgraph already has a node representing this exact same process. We'd like to de-duplicate this process node.
+
+**Output:** TODO
+
+
+### Graph Merger
+**Work:** Write the new edges and nodes from Node Identifier to Dgraph.
+
+**Output:** TODO (it looks like exactly the same as node identifier's output, just re-encoded?)
+
+
+### Analyzer Dispatcher
+**Work:** Determines which analyzers to execute against recently-merged graphs
+
+**Output:** TODO
+
+
+### Analyzer Executor
+**Work:** Executes queries against Dgraph, and - upon a successful query hit - determines which lenses and risk nodes to create
+
+**Output:** A JSON dump of risks and lenses to be created by the Engagement Creator (see `analyzer_executor.py#emit_event` for specifics)
+
+
+### Engagement Creator
+**Work:** Simply writes the risks and lenses described in its input to Dgraph.
```

Comment:
maybe s/write/append/ to describe the semantics (doesn't mutate existing structure, if that's correct)